src/pages/gd/Stake/Savings/index.tsx

```diff
@@ -22,24 +22,27 @@ const SavingsDeposit = styled.div`
     margin-top: 10px;
 `
 
-export const Savings = (): JSX.Element => {
+export const Savings = ({ requiredChain }: { requiredChain: number }): JSX.Element => {
     const [isOpen, setIsOpen] = useState(false)
     const { account, chainId } = useActiveWeb3React()
-    const { stats, error } = useSavingsStats(10)
+    const { stats, error } = useSavingsStats(requiredChain, 10)
     const { i18n } = useLingui()
-    const toggleModal = useCallback(() => setIsOpen(!isOpen), [setIsOpen, isOpen])
     const { width } = useWindowSize()
     const isMobile = width ? width <= 768 : undefined
     const { defaultEnv } = useGetEnvChainId()
-    const g$ = G$(chainId, defaultEnv)
-    const getData = sendGa
+    const g$ = G$(requiredChain, defaultEnv)
 
     useEffect(() => {
         if (error) {
             console.error('Unable to fetch global stats:', { error })
         }
     }, [error])
 
+    const toggleModal = useCallback(() => {
+        sendGa({ event: 'savings', action: 'savingsStart' })
+        setIsOpen(!isOpen)
```

Comment:
you don't need refer to isOpen
useState setter could accept callback receiving current value like it was with setState (see my toggle function as example)

setIsOpen(isOpen => !isOpen)

so you doun;t need isOpen in the deps